rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -537,7 +537,74 @@ object Reduce {
+              case (GString(keyString), GString(valueString)) =>
+                Applicative[M].pure[(String, String)](keyString -> valueString)
+              case (GString(keyString), GInt(valueInt)) =>
+                Applicative[M].pure[(String, String)](keyString -> valueInt.toString)
+              // TODO: Add cases for other ground terms as well? Maybe it would be better
+              // to implement cats.Show for all ground terms.
+              case (_: GString, value) =>
+                s.raiseError[(String, String)](
+                  ReduceError(s"Error: interpolation doesn't support ${value.getClass}")
+                )
+              case _ =>
+                s.raiseError[(String, String)](
+                  ReduceError("Error: interpolation Map should only contain String keys")
+                )
+            }
+          for {
+            v1 <- evalSingleExpr(p1.get)
+            v2 <- evalSingleExpr(p2.get)
+            result <- (v1.exprInstance, v2.exprInstance) match {
+                       case (GString(lhs), EMapBody(ParMap(rhs, _, _))) =>
+                         rhs.iterator
+                           .map {
+                             case (k, v) =>
+                               for {
+                                 keyExpr   <- evalSingleExpr(k)
+                                 valueExpr <- evalSingleExpr(v)
+                                 result    <- evalToStringPair(keyExpr, valueExpr)
+                               } yield result
+                           }
+                           .toList
+                           .sequence[M, (String, String)]
+                           .map { keyValuePairs =>
+                             val result  = StringBuilder.newBuilder
+                             var current = lhs
+                             while (current.nonEmpty) {
+                               keyValuePairs.find {
+                                 case (k, v) => current.startsWith("${" + k + "}")
+                               } match {
+                                 case Some((k, v)) =>
+                                   result ++= v
+                                   current = current.drop(k.length + 3)
+                                 case None =>
+                                   result += current.head
+                                   current = current.tail
+                               }
+                             }
+                             GString(result.toString)
+                           }
+                       case (_: GString, _) =>
+                         s.raiseError(ReduceError("Error: String can be interpolated only by Map"))
+                       case _ =>
+                         s.raiseError(ReduceError("Error: only Strings can be interpolated"))
+                     }
+          } yield result
+        case EPlusPlusBody(EPlusPlus(p1, p2)) =>
+          for {
+            v1 <- evalSingleExpr(p1.get)
```

Comment:
I think you can drop `.get` . There are some residues here and there but thanks to 
```
Par p1 = 1 [(scalapb.field).no_box = true];
```

in proto files it won't wrap in `Option`.

Reply:
Are you sure? Every single case here has them. Should I also delete them as well?

Reply:
If it compiles after removing then it will work.